### PR TITLE
 pre-release ` tm_data_table`  document update

### PR DESCRIPTION
**Pull request by kartikeyakirar**
part of https://github.com/insightsengineering/teal.modules.general/issues/624

### CODE REVIEW

File: R/tm_data_table.R [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
 #'
-#' A data table viewer shows the data using a paginated table.
-#' specifically designed for use with `data.frames`.
-#' @md
+#' Module provides a dynamic and interactive way to view `data.frame`s in a `teal` application.
+#' It uses the `DT` package to display data tables in a paginated, searchable, and sortable format,
+#' which helps to enhance data exploration and analysis.
+#'
+#' The `DT` package has an option `DT.TOJSON_ARGS` to show `Inf` and `NA` in data tables.
+#' configure the `DT.TOJSON_ARGS` option via
```
Comment (m7pr) [inline R/tm_data_table.R:8]:
```suggestion
#' Configure the `DT.TOJSON_ARGS` option via
```

File: R/tm_data_table.R [hunk 2]
```diff
@@ -1,34 +1,34 @@
-#' Data Table Viewer `teal` Module
+#' Data table viewer module
 #'
-#' A data table viewer shows the data using a paginated table.
-#' specifically designed for use with `data.frames`.
-#' @md
+#' Module provides a dynamic and interactive way to view `data.frame`s in a `teal` application.
+#' It uses the `DT` package to display data tables in a paginated, searchable, and sortable format,
```
Comment (m7pr) [inline R/tm_data_table.R:4]: Maybe a link do `DT` package?

File: R/tm_data_table.R [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
-#'   shown and in what order. Names in the vector have to correspond with datasets names.
-#'   If vector of length zero (default) then all datasets are shown.
-#'   Note: Only datasets of the `data.frame` class are compatible;
-#'   using other types will cause an error.
-#' @param dt_args (named `list`) Additional arguments to be passed to `DT::datatable`
-#'   (must not include `data` or `options`).
-#' @param dt_options (named `list`) The `options` argument to `DT::datatable`. By default
-#'   `list(searching = FALSE, pageLength = 30, lengthMenu = c(5, 15, 30, 100), scrollX = TRUE)`
+#' shown and in what order. Names in the vector have to correspond with datasets names.
+#' If vector of `length==0` (default) then all datasets are shown.
```
Comment (m7pr) [inline R/tm_data_table.R:21]:
```suggestion
#' If vector of `length == 0` (default) then all datasets are shown.
```

File: R/tm_data_table.R [hunk 4]
```diff
@@ -127,8 +127,9 @@ tm_data_table <- function(label = "Data Table",
   )
 }
 
-
-# ui page module
+# UI page module.
```
Comment (m7pr) [inline R/tm_data_table.R:130]: You do not have dots at the end of comments in this PR
https://github.com/insightsengineering/teal.modules.general/pull/653/files
Please unify
```suggestion
# UI page module
```

File: R/tm_data_table.R [hunk 5]
```diff
@@ -162,8 +163,9 @@ ui_page_data_table <- function(id,
   )
 }
 
-
-# server page module
+# server page module.
```
Comment (m7pr) [inline R/tm_data_table.R:166]: You do not have dots at the end of comments in this PR
https://github.com/insightsengineering/teal.modules.general/pull/653/files
Please unify
```suggestion
# server page module
```

File: R/tm_data_table.R [hunk 6]
```diff
@@ -247,6 +249,9 @@ srv_page_data_table <- function(id,
   })
 }
 
+# UI function for the data_table module.
```
Comment (m7pr) [inline R/tm_data_table.R:252]: ?
```suggestion
# UI function for the data_table module
```

File: R/tm_data_table.R [hunk 7]
```diff
@@ -276,6 +281,9 @@ ui_data_table <- function(id,
   )
 }
 
+# Server function for the data_table module.
```
Comment (m7pr) [inline R/tm_data_table.R:284]: ?
```suggestion
# Server function for the data_table module
```

**Comment by m7pr** [comment]
@kartikeyakirar I left you a few comments

File: R/tm_data_table.R (hunk 2 above)
Comment (kartikeyakirar) [reply R/tm_data_table.R:4]: I have linked the functions from `DT` that are mentioned in the parameter description. In description we did not link package directly (functions or topic only)

Comment (kartikeyakirar) [reply R/tm_data_table.R:4]: [25b47e4](https://github.com/insightsengineering/teal.modules.general/pull/667/commits/25b47e40d945bcfc97e4be841ae69ffdb710b985)

File: R/tm_data_table.R (hunk 5 above)
Comment (kartikeyakirar) [reply R/tm_data_table.R]: done. removed `.`

**Review by m7pr — APPROVED** [review]

**Pull request merged**